services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/resource_tracker_background_task.py

```diff
@@ -94,7 +95,9 @@ async def _close_unhealthy_service(
         return
 
     # 2. Close the billing transaction (as not billed)
-    if running_service.wallet_id and running_service.pricing_unit_cost:
+    if running_service.wallet_id:
```

Comment:
Currently, it is coupled with wallet_id, if there is wallet_id this implies there will be pricing_unit_cost otherwise both are None. But you are right, `running_service.pricing_unit_cost is not None` would also do the job.